utils/vault_client.py

```diff
@@ -48,3 +48,15 @@ def read(path, field):
         raise SecretFieldNotFound("{}/{}".format(path, field))
 
     return secret_field
+
+
+def read_all(path):
+    global _client
+    init_from_config()
+
+    secret = _client.read(path)
+
+    if secret is None or 'data' not in secret:
+        raise SecretNotFound(path)
```

Comment:
minor pet-peeve, but I kinda prefer try/except KeyError

Reply:
Plus, I was going for compliance:

https://github.com/app-sre/qontract-reconcile/blob/2237e286aef2ba6cc3cf746d3b204b8646292472/utils/vault_client.py#L42

would you suggest we change the `read` function as well?

Reply:
Does this make more sense?

```
secret = _client.read(path)

    if secret is None:
        raise SecretNotFound(path)

    try:
        return secret['data']
    except KeyError:
        raise SecretHasNoData(path)
```